text/0000-optimise-attr.md

```diff
@@ -0,0 +1,161 @@
+- Feature Name: optimize_attr
+- Start Date: 2018-03-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC introduces the `#[optimize]` attribute for controlling optimisation level on a per-item
```

Comment:
This is a *massive* nit, but given that the attribute is spelled `optimize` for consistency with precedent and expectations from other tools and environments, could you please use that same spelling consistently in the RFC? And, in particular, in the filename (currently `0000-optimise-attr.md`)?

Also, given that it seems likely to become an issue, could you explicitly mention, somewhere in the RFC, that the spelling "optimize" was chosen specifically based on precedent and expectations from other tools and environments, and that the implementation should include a test to confirm that `optimise` produces a clear error message that points people to the correct spelling (including a rustfix)?

Reply:
I’ll try to fix spelling before a merge.

> implementation should include a test

I feel that it would be sufficient to require such a test in the tracking issue. 

> could you explicitly mention, somewhere in the RFC

Seems like a difficult thing to fit into the RFC without it sticking out like a sore thumb. Not sure if it is necessary either, given that `-ize` spelling is prefered across the compiler anyway.

Reply:
You already have an "Rationale and alternatives" section; that seems like the right place for a one-paragraph mention.

I'm mostly mentioning it here so it doesn't get lost. I wouldn't be surprised if Rust's existing spelling-correction mechanisms catch this, which just makes it a matter of a single test case.

Reply:
Do we have more attributes of the form `-ize`?  If so, we could do a more general version of the error message, or just employ the usual levenshtein distance on attribute names...

Reply:
`-ize` in general, not just attributes, but even in attributes there’s not a single hit for `-ise`, but a few for `-ize`:

* `#[sanitizer_runtime]` (rather than `#[sanitiser_runtime]`);
* `#[derive(Deserialize)]` (instead of `#[derive(Deserialise)]`, although that’s a library type);
* [`parenthesized_params_in_types_and_modules`](https://github.com/rust-lang/rust/issues/42238) lint.

Multiple CLI flags use `-ize` spelling as well.